test/predicator_test.exs

```diff
@@ -5,182 +5,439 @@ defmodule PredicatorTest do
 
   @moduletag :parsing
 
+  describe "BOOLEAN" do
+    test "commpiles string key instructions" do
+      assert {:ok, [["load", "foo"], ["to_bool"]]} = Predicator.compile("foo")
+      assert {:ok, [["load", "foo"], ["to_bool"], ["not"]]} = Predicator.compile("!foo")
+      assert {:ok, [["lit", true]]} = Predicator.compile("true")
+      assert {:ok, [["lit", false]]} = Predicator.compile("false")
+    end
+
+    test "compiles atom key instructions" do
+      assert {:ok, [[:load, :foo], [:to_bool]]} = Predicator.compile("foo", :atom_key_inst)
+
+      assert {:ok, [[:load, :foo], [:to_bool], [:not]]} =
+               Predicator.compile("!foo", :atom_key_inst)
+
+      assert {:ok, [[:lit, true]]} = Predicator.compile("true", :atom_key_inst)
+      assert {:ok, [[:lit, false]]} = Predicator.compile("false", :atom_key_inst)
+      assert {:ok, [[:lit, true], [:not]]} = Predicator.compile("!true", :atom_key_inst)
+      assert {:ok, [[:lit, false], [:not]]} = Predicator.compile("!false", :atom_key_inst)
+    end
+
+    test "evaluates to true" do
+      assert Predicator.matches?("true") == true
+      assert Predicator.matches?("!false") == true
+      assert Predicator.matches?("foo", foo: true) == true
+      assert Predicator.matches?("!foo", foo: false) == true
+    end
+
+    test "evaluates to false" do
+      assert Predicator.matches?("!true") == false
+      assert Predicator.matches?("false") == false
+      assert Predicator.matches?("foo", foo: false) == false
+      assert Predicator.matches?("!foo", foo: true) == false
+    end
+  end
+
   describe "BETWEEN" do
-    test "compiles" do
+    test "compiles atom key instructions" do
       assert {:ok, [[:load, :age], [:lit, 5], [:lit, 10], [:comparator, :BETWEEN]]} =
                Predicator.compile("age between 5 and 10", :atom_key_inst)
 
-      assert {:ok, [["load", "age"], ["lit", 5], ["lit", 10], ["comparator", "BETWEEN"]]} =
-               Predicator.compile("age between 5 and 10")
+      assert {:ok, [[:load, :age], [:lit, 5], [:lit, 10], [:comparator, :BETWEEN], [:not]]} =
+               Predicator.compile("!age between 5 and 10", :atom_key_inst)
+    end
+
+    test "compiles string key instructions" do
+      assert {:ok, [["load", "age"], ["lit", 5], ["lit", 10], ["comparator", "BETWEEN"], ["not"]]} =
+               Predicator.compile("!age between 5 and 10")
```

Comment:
Added the `!` possibility to every type of predicate 😎 more tests the better, right? 😅